File: simplified-books-core/src/main/java/org/nypl/simplified/books/core/AccountGetCachedCredentialsListenerType.java
package org.nypl.simplified.books.core;

/**
 * The type of listeners for receiving cached credentials.
 */

public interface AccountGetCachedCredentialsListenerType
{
  /**
   * The account is not logged in.
   */

  void onAccountIsNotLoggedIn();

  /**
   * The account is logged in with the given credentials.
   *
   * @param credentials The current credentials
   */

  void onAccountIsLoggedIn(
    AccountCredentials credentials);
}

File: simplified-books-core/src/main/java/org/nypl/simplified/books/core/AccountGetCachedCredentialsListenerType.kt
package org.nypl.simplified.books.core

/**
 * The type of listeners for receiving cached credentials.
 */

interface AccountGetCachedCredentialsListenerType {

  /**
   * The account is not logged in.
   */

  fun onAccountIsNotLoggedIn()

  /**
   * The account is logged in with the given credentials.
   *
   * @param credentials The current credentials
   */

  fun onAccountIsLoggedIn(credentials: AccountCredentials)

}
